infra/bisector.py

```diff
@@ -0,0 +1,175 @@
+      '--test_case', help='the test_case to be reproduced', required=True)
+  parser.add_argument('--engine', default='libfuzzer')
+  parser.add_argument(
+      '--sanitizer',
+      default='address',
+      help='the default is "address"')
+  parser.add_argument('--architecture', default='x86_64')
+  args = parser.parse_args()
+
+  if not os.getcwd().endswith('oss-fuzz'):
+    print("Error: bisector.py needs to be run from the oss-fuzz home directory")
+    return 1
+
+  error_sha = bisect(args.project_name, args.commit_old,
+                             args.commit_new, args.engine, args.sanitizer,
+                             args.architecture, args.test_case,
+                             args.fuzz_target)
+  if not error_sha:
+    print('No error was found in commit range %s:%s' %
+          (args.commit_old, args.commit_new))
+  else:
+    print('Error was introduced at commit %s' % error_sha)
+
+
+def bisect(project_name, commit_old, commit_new, engine, sanitizer,
+                   architecture, test_case, fuzz_target):
+  """Creates an bisection call that kicks off the error detection.
+
+  This function is necessary in order to get the error code of the newest commit. This
+  sets a standard for what the error actually is.
+
+  Args:
+    project_name: The name of the oss fuzz project that is being checked
+    commit_old: The oldest commit in the error regression range
+    commit_new: The newest commit in the error regression range
+    engine: The fuzzing engine to be used
+    sanitizer: The fuzzing sanitizer to be used
+    architecture: The system architecture being fuzzed
+    test_case: The file path of the test case that triggers the error
+    fuzz_target: The name of the fuzzer to be tested
+
+  Returns:
+    The commit SHA that introduced the error or None
+  """
+  local_store_path = 'tmp'
+  repo_url = infer_main_repo(project_name, local_store_path, commit_old)
+  repo_manager = RepoManager(repo_url, local_store_path)
+  commit_list = repo_manager.get_commit_list(commit_old, commit_new)
+  build_fuzzer_from_commit(project_name, commit_list[0], repo_manager.repo_dir,
+                           engine, sanitizer, architecture, repo_manager)
+  error_code = reproduce_impl(project_name, fuzz_target, False, [], [],
+                              test_case)
+  # Handle the case where there is only one SHA passed in
+  if len(commit_list) == 1:
+    if not error_code:
+      return None
+    return commit_list[0]
+  index = _bisection(project_name, 0,
+                    len(commit_list) - 1, commit_list, repo_manager,
+                    len(commit_list), error_code, engine, sanitizer,
+                    architecture, test_case, fuzz_target)
+  if index is not None:
+    return commit_list[index]
+  return None
+
+
+def _bisection(project_name, commit_new_idx, commit_old_idx, commit_list,
+              repo_manager, last_error_idx, error_code, engine, sanitizer,
+              architecture, test_case, fuzz_target):
```

Comment:
nit: please call this "testcase" to be consistent with other uses of the word.
(I realize "testcase" is not a real word but I think consistency is more important here).